### PR TITLE
[dask] Support more meta data.

### PR DESCRIPTION
**Pull request by trivialfis**
* label_lower_bound
* label_upper_bound
* base_margin

### CODE REVIEW

**Comment by trivialfis** [comment]
@hcho3 I modified the test for survival training to use it for dask.  Could you please help taking a look?

**Review by RAMitchell — APPROVED** [review]

File: python-package/xgboost/dask.py [hunk 1]
```diff
@@ -191,6 +191,9 @@ def __init__(self,
                  label=None,
                  missing=None,
                  weight=None,
+                 base_margin=None,
+                 label_lower_bound=None,
+                 label_upper_bound=None,
```
Comment (hcho3) [inline python-package/xgboost/dask.py:196]: These new parameter should be added to the docstring.

File: tests/python/test_with_dask.py [hunk 2]
```diff
@@ -482,16 +481,59 @@ def test_predict():
...
-            shap = xgb.dask.predict(client, model=booster, data=dtrain, pred_contribs=True)
+            shap = xgb.dask.predict(client, model=booster, data=dtrain,
+                                    pred_contribs=True)
             assert shap.ndim == 2
             assert shap.shape[0] == kRows
             assert shap.shape[1] == kCols + 1
 
 
+def test_aft_survival():
+    # survival doesn't handle empty dataset well.
```
Comment (hcho3) [inline tests/python/test_with_dask.py:497]: I did not know this. How can we fix this?

**Review by hcho3 — COMMENTED** [review]
Thanks for adding survival analysis in the Dask test. Some comments:

File: tests/python/test_with_dask.py (hunk 2 above)
Comment (trivialfis) [reply tests/python/test_with_dask.py:497]: @hcho3 There are checks:
``` c++
    CHECK_NE(info.labels_lower_bound_.Size(), 0U)
      << "labels_lower_bound cannot be empty";
    CHECK_NE(info.labels_upper_bound_.Size(), 0U)
      << "labels_upper_bound cannot be empty";
```

in `survival_metric.cu::EvalEWiseSurvivalBase::Eval`.

Comment (hcho3) [reply tests/python/test_with_dask.py:497]: Should we remove the checks and have the objective return zero?

Comment (trivialfis) [reply tests/python/test_with_dask.py:497]: Depends on the synchronization I think.

Comment (hcho3) [reply tests/python/test_with_dask.py:497]: AFT is an element-wise objective, so it should behave like `reg:squarederror` when it comes to handling empty datasets. Let me create a new issue for this.

Comment (hcho3) [reply tests/python/test_with_dask.py:497]: Opened #6139

**Review by hcho3 — APPROVED** [review]

**Pull request merged**